x-pack/plugins/infra/public/pages/metrics/inventory_view/components/survey_kubernetes.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiButton, EuiFlexGroup, EuiFlexItem, EuiGlobalToastList } from '@elastic/eui';
+import React, { useEffect, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useWaffleOptionsContext } from '../hooks/use_waffle_options';
+
+const KUBERNETES_TOAST_STORAGE_KEY = 'kubernetesToastKey';
+const KUBERNETES_FEEDBACK_LINK = 'https://ela.st/k8s-feedback';
+
+const openInNewTab = (url: string) => {
+  window.open(url, '_blank', 'noreferrer');
+};
+
+export const SurveyKubernetes = () => {
+  const { nodeType } = useWaffleOptionsContext();
+  const podNodeType: typeof nodeType = 'pod';
+
+  const [isToastSeen, setIsToastSeen] = useState(() => {
+    const initialState = localStorage.getItem(KUBERNETES_TOAST_STORAGE_KEY);
+    return initialState === 'true';
+  });
+
+  useEffect(() => {
+    localStorage.setItem(KUBERNETES_TOAST_STORAGE_KEY, isToastSeen ? 'true' : 'false');
+  }, [isToastSeen]);
```

Comment:
**suggestion**: to keep the code declarative, you could use the `useLocalStorage` hook to delegate this piece of logic that saves/retrieves the value from the localStorage:
```js
import useLocalStorage from 'react-use/lib/useLocalStorage';

const [isToastSeen, setIsToastSeen] = useLocalStorage(KUBERNETES_TOAST_STORAGE_KEY, false);
```

Reply:
@tonyghiani Thank you for the good suggestions! I applied them ✅